models/articles.models.js

```diff
@@ -1,14 +1,49 @@
 const db = require('../db/connection');
+const { checkTopicSlugExists } = require('./topics.models');
 
-exports.selectAllArticles = () => {
-  return db.query(
+exports.selectAllArticles = (topic, sort_by = 'created_at', order = 'desc') => {
+  const validSortByOptions = [
+    'article_id',
+    'title',
+    'topic',
+    'author',
+    'created_at',
+    'votes',
+    'article_img_url',
+    'comment_count'
+  ];
+
+  if (!validSortByOptions.includes(sort_by)) {
+    return Promise.reject({ status: 400, message: 'Invalid sort_by query' });
+  }
+
+  if (!['asc', 'desc'].includes(order)) {
```

Comment:
maybe worth adding capitals as well? "ASC, DESC"

Reply:
Would the below achieve the same result and would it warrant a test? Perhaps I could test for 'asc' and 'DESC' instead of the 2 current lowercase tests.

```javascript
if (!['asc', 'desc'].includes(order.toLowerCase()))
```